### PR TITLE
otel-cli: init at 0.0.19

### PR DESCRIPTION
**Pull request by emattiza**
Initial package addition of otel-cli, a cli opentelemetry application to
support command-line application instrumentation and observability.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

#143471

### CODE REVIEW

File: pkgs/tools/misc/otel-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    ln -s ${getent}/bin/getent /bin/getent
+    # Must build full binary cli app for test run
+    go build .
+  '';
+  doCheck = false;
+  meta = with lib; {
+    homepage = "https://github.com/equinix-labs/otel-cli";
+    description = "a cli tool for otel instrumenting command line work";
+    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux"];
+    license = with licenses; [asl20];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:26]:
```suggestion
    license = with licenses; [ asl20 ];
```

File: pkgs/tools/misc/otel-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  checkInputs = [ getent ];
+  vendorSha256 = "sha256-IJ2Gq5z1oNvcpWPh+BMs46VZMN1lHyE+M7kUinTSRr8=";
+  preCheck = ''
+    # This is necessary due to path hard-coding in the E2E Acceptance Test
+    ln -s ${getent}/bin/getent /bin/getent
+    # Must build full binary cli app for test run
+    go build .
+  '';
+  doCheck = false;
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:22]:
```suggestion
  '';

  doCheck = false;

  meta = with lib; {
```
Why are tests disabled?

File: pkgs/tools/misc/otel-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0.0.19";
+  src = fetchFromGitHub {
+    owner = "equinix-labs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-jQKoj55ty68yMDOa7/ulH1xZdI1GUoFyiPWdBHRWDNM=";
+  };
+  checkInputs = [ getent ];
+  vendorSha256 = "sha256-IJ2Gq5z1oNvcpWPh+BMs46VZMN1lHyE+M7kUinTSRr8=";
+  preCheck = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:15]:
```suggestion
buildGoModule rec {
  pname = "otel-cli";
  version = "0.0.19";

  src = fetchFromGitHub {
    owner = "equinix-labs";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-jQKoj55ty68yMDOa7/ulH1xZdI1GUoFyiPWdBHRWDNM=";
  };

  checkInputs = [ getent ];

  vendorSha256 = "sha256-IJ2Gq5z1oNvcpWPh+BMs46VZMN1lHyE+M7kUinTSRr8=";

  preCheck = ''
```

**Review by ghuntley — APPROVED** [review]

File: pkgs/tools/misc/otel-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub, getent }:
+
+buildGoModule rec {
+  pname = "otel-cli";
+  version = "0.0.19";
```
Comment (ghuntley) [inline pkgs/tools/misc/otel-cli/default.nix:5]: `0.0.20`

File: pkgs/tools/misc/otel-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "equinix-labs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-jQKoj55ty68yMDOa7/ulH1xZdI1GUoFyiPWdBHRWDNM=";
```
Comment (ghuntley) [inline pkgs/tools/misc/otel-cli/default.nix:11]: `1z2a0n64gszm098dlq64iy7fnj54l8f0mksdvn42sp4nan6gym4m`

File: pkgs/tools/misc/otel-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  checkInputs = [ getent ];
+
+  vendorSha256 = "sha256-IJ2Gq5z1oNvcpWPh+BMs46VZMN1lHyE+M7kUinTSRr8=";
```
Comment (ghuntley) [inline pkgs/tools/misc/otel-cli/default.nix:16]: `sha256-S2uy57R9W5CJtL8M6kYyaOaRXlUGtCPojvyjG8XT/G0=`

**Review by ghuntley — COMMENTED** [review]
Good to merge as is. If you wanted to bump then here are the values for 0.0.20

File: pkgs/tools/misc/otel-cli/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+  preCheck = ''
+    # This is necessary due to path hard-coding in the E2E Acceptance Test
+    ln -s ${getent}/bin/getent /bin/getent
+    # Must build full binary cli app for test run
+    go build .
+  '';
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:25]: if doCheck is false then we don't run any of the tests.

File: pkgs/tools/misc/otel-cli/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/equinix-labs/otel-cli";
+    description = "a cli tool for otel instrumenting command line work";
+    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux"];
+    license = with licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ emattiza ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ emattiza ];
```

**Pull request closed without merging**